src/Core/Framework/DependencyInjection/CompilerPass/RepositoryAutowireCompilerPass.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\DependencyInjection\CompilerPass;
+
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use function mb_strstr;
+
+/**
+ * This CompilerPass filters for service ids ending on `.repository`, adding an alias to the container
+ * for service arguments with the type `EntityRepositoryInterface`
+ */
+class RepositoryAutowireCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->getServiceIds() as $serviceId) {
+            if (mb_strstr($serviceId, '.repository') === false) {
```

Comment:
I don't like the solution. The check is too open. Maybe Do this in the DAL Compiler Pass where .repository will be registered. So we are sure that, they get only recognized

Reply:
Good idea, i'll fix that!

Reply:
The Alias is now registered where the normal service is registered as well, thanks for the pointer! :blue_heart: